explore/future_compat.h

```diff
@@ -62,12 +62,16 @@
     #define VW_WARNING_DISABLE_DEPRECATED_USAGE    VW_WARNING_DISABLE(4996)
 #  define VW_WARNING_DISABLE_BADLY_FORMED_XML VW_WARNING_DISABLE(4635)
 #  define VW_WARNING_DISABLE_CPP_17_LANG_EXT VW_WARNING_DISABLE(4984)
```

Comment:
It looks at least from first glance that we don't need this suppression if we are using the language level detecting macro https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-c4984?view=msvc-170